tenant/http_server_bucket.go

```diff
@@ -149,25 +132,42 @@ func newBucket(pb *influxdb.Bucket) *bucket {
 		Name:                pb.Name,
 		Description:         pb.Description,
 		RetentionPolicyName: pb.RetentionPolicyName,
-		RetentionRules:      rules,
+		RetentionRules:      []retentionRule{rule},
 		CRUDLog:             pb.CRUDLog,
 	}
 }
 
+type retentionRuleUpdate struct {
+	Type                      string `json:"type"`
+	EverySeconds              *int64 `json:"everySeconds"`
+	ShardGroupDurationSeconds *int64 `json:"shardGroupDurationSeconds"`
+}
+
 // bucketUpdate is used for serialization/deserialization with retention rules.
 type bucketUpdate struct {
-	Name           *string         `json:"name,omitempty"`
-	Description    *string         `json:"description,omitempty"`
-	RetentionRules []retentionRule `json:"retentionRules,omitempty"`
+	Name           *string               `json:"name,omitempty"`
+	Description    *string               `json:"description,omitempty"`
+	RetentionRules []retentionRuleUpdate `json:"retentionRules,omitempty"`
 }
 
 func (b *bucketUpdate) OK() error {
+	if len(b.RetentionRules) > 1 {
+		return &influxdb.Error{
+			Code: influxdb.EUnprocessableEntity,
+			Msg:  "buckets cannot have more than one retention rule at this time",
+		}
+	}
+
 	if len(b.RetentionRules) > 0 {
-		_, err := b.RetentionRules[0].RetentionPeriod()
-		if err != nil {
-			return err
+		rule := b.RetentionRules[0]
+		if rule.EverySeconds != nil && *rule.EverySeconds < 0 {
```

Comment:
Should we check Shard Group Duration is non-negative as well?

Reply:
Done, and added tests for it.